### PR TITLE
ByteBufferPool bookkeeping enhancements

### PR DESCRIPTION
**Pull request by JonathanLennox**

### CODE REVIEW

File: src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java [hunk 1]
```diff
@@ -113,6 +127,19 @@ private static long threadId()
         return Thread.currentThread().getStackTrace();
     }
 
+    /**
+     * Gets a stack trace as a multi-line string.
+     */
+    private static String stackTraceAsString(StackTraceElement[] stack)
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java]: While you're doing this, can you write a kotlin one which is inlined so we don't get these methods as part of the stack?  I'm fine with just doing it in JVB.

Comment (JonathanLennox) [reply src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java]: I can use `UtilKt.getStackTrace()` but Java won't inline Kotlin, so they still show up on the stack.  Probably still worth it, though.

File: src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java [hunk 2]
```diff
@@ -198,10 +228,23 @@ else if (len < 2000)
...
+            {
+                logger.info("Thread " + threadId()
+                    + " returned a previously-returned buffer at\n"
+                    + getStackTraceAsString() +
+                    "previously returned at\n" +
+                    stackTraceAsString(returnedBookkeeping.get(arrayId).deallocTrace));
+            }
+            else
             {
                 logger.info("Thread " + threadId()
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java]: As I'm doing debugging, I'm finding I wish we had the size of the buffer as well here.

File: src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java (hunk 1 above)
Comment (bbaldino) [reply src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java]: Ah, bummer.

File: src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java [hunk 3]
```diff
@@ -198,14 +208,27 @@ else if (len < 2000)
...
+                returnedBookkeeping.put(arrayId, new ReturnedBufferBookkeepingInfo(bookkeeping.get(arrayId), UtilKt.getStackTrace()));
+                bookkeeping.remove(arrayId);
+            }
+            else if (returnedBookkeeping.containsKey(arrayId))
+            {
+                logger.info("Thread " + threadId()
+                    + " returned a previously-returned buffer at\n"
+                    + UtilKt.getStackTrace() +
+                    "previously returned at\n" +
+                    returnedBookkeeping.get(arrayId).deallocTrace);
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java]: I've seen a race happen here where this causes an NPE...so we should check and save the entry in one step.

**Pull request merged**